custom/icds/data_management/base.py

```diff
@@ -0,0 +1,53 @@
+from corehq.form_processor.backends.sql.dbaccessors import CaseReindexAccessor
+from corehq.util.doc_processor.interface import BulkDocProcessor
+from corehq.util.doc_processor.sql import SqlDocumentProvider
+
+
+class DataManagement(object):
+    slug = ""
+    name = ""
+    case_type = None
+    doc_processor = None
+
+    def __init__(self, domain, *args, **kwargs):
+        self.domain = domain
+
+    def case_accessor(self):
```

Comment:
why is this a function and not a class property like the other fields?

Reply:
You mean something like this?
```
class DataManagement(object):
    case_accessor = None

class SQLBasedDataManagement(DataManagement):
    case_accessor = CaseReindexAccessor

    def run(self, iteration_key):
        record_provider = SqlDocumentProvider(iteration_key, self.case_accessor(
            domain=self.domain,
            case_type=self.case_type,
            limit_db_aliases=self.db_aliases,
            start_date=self.from_date,
            end_date=self.till_date
        ))
```

Considering the arguments passed, It's going to change for each class and each instance of class. Keeping it on instance made `run` method generic and i could keep that at one place. 
I can still keep it as a property on instance, but considered not caching it because its not reused, and its dependent on the properties set on the instant which can change if someone is running these requests manually